GitCommands/Git/AheadBehindDataProvider.cs

```diff
@@ -64,22 +69,39 @@ public AheadBehindDataProvider(Func<IExecutable> getGitExecutable)
             }
 
             var matches = _aheadBehindRegEx.Matches(result);
-            if (matches.Count < 1 || (matches.Count == 1 && (!matches[0].Groups["branch"].Success ||
-                                                             !(matches[0].Groups["ahead_p"].Success || matches[0].Groups["ahead_u"].Success ||
-                                                               matches[0].Groups["behind_p"].Success || matches[0].Groups["behind_u"].Success))))
-            {
-                return null;
-            }
-
             var aheadBehindForBranchesData = new Dictionary<string, AheadBehindData>();
             foreach (Match match in matches)
             {
+                var branch = match.Groups["branch"].Value;
+                var remoteRef = (match.Groups["remote_p"].Success && !string.IsNullOrEmpty(match.Groups["remote_p"].Value))
+                            ? match.Groups["remote_p"].Value
+                            : match.Groups["remote_u"].Value;
+                if (string.IsNullOrEmpty(branch) || string.IsNullOrEmpty(remoteRef))
+                {
+                    continue;
+                }
+
                 aheadBehindForBranchesData.Add(match.Groups["branch"].Value,
                     new AheadBehindData
                     {
-                        // The information is displayed in the push button, so the push info  is preferred (may differ from upstream)
-                        Branch = match.Groups["branch"].Value,
-                        AheadCount = match.Groups["ahead_p"].Success ? match.Groups["ahead_p"].Value : match.Groups["ahead_u"].Value,
+                        // The information is displayed in the push button, so the push info is preferred (may differ from upstream)
+                        Branch = branch,
+                        AheadCount =
+
+                            // Prefer push to upstream for the count
+                            match.Groups["ahead_p"].Success
+                            ? match.Groups["ahead_p"].Value
+                            : match.Groups["ahead_u"].Success
+                            ? match.Groups["ahead_u"].Value
+
+                            // No information about the remote branch, it is gone
+                            : match.Groups["gone_p"].Success || match.Groups["gone_u"].Success
+                            ? AheadBehindData.Gone
+
+                            // A remote exists, but "track" does not display the count if ahead/behind match
+                            : "0",
+
```

Comment:
```suggestion
                        AheadCount =
                            // Prefer push to upstream for the count
                            match.Groups["ahead_p"].Success
                            ? match.Groups["ahead_p"].Value
                            : match.Groups["ahead_u"].Success
                              ? match.Groups["ahead_u"].Value
                              : // No information about the remote branch, it is gone
                                match.Groups["gone_p"].Success || match.Groups["gone_u"].Success
                                ? AheadBehindData.Gone
                                : // A remote exists, but "track" does not display the count if ahead/behind match
                                  "0",
```

Reply:
There must be a newline before a comment....

Reply:
Right now it is not very readable... You can suppress the warning using #pragma

Reply:
> There must be a newline before a comment....

Or use `////`.

Reply:
I used #pragma instead of misusing ////
Even more conditions now, I prefer to not right shift (should be 4 spaces if done)